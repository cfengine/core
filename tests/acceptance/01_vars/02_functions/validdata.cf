###########################################################
#
# Test validdata() with valid and invalid data files
#
###########################################################

body common control
{
    inputs => { "../../default.cf.sub" };
    bundlesequence => { default($(this.promise_filename)) };
    version => "1.0";
}

###########################################################

bundle agent test
{
  vars:
    "valid_json" string => "should appear",
      if => validdata(readfile("$(this.promise_filename).json", inf), "JSON");
    "invalid_json" string => "should not appear",
      if => validdata(readfile("$(this.promise_filename).inv.json", inf), "JSON");

    "valid_validjson" string => "should appear",
      if => validjson(readfile("$(this.promise_filename).json", inf));
    "invalid_validjson" string => "should not appear",
      if => validjson(readfile("$(this.promise_filename).inv.json", inf));
}

###########################################################

bundle agent check
{
  methods:
    "check" usebundle => dcs_check_state(test,
                                         "$(this.promise_filename).expected.json",
                                         $(this.promise_filename));
}
