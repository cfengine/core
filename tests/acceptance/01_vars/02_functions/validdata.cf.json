{
	"test": [1,2,3,4],
	"test2": {
		"test3": 1
	},
	"test3": "Yes"
}
