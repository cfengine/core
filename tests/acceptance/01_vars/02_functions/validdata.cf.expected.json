{
  "valid_json": "should appear",
  "valid_validjson": "should appear"
}
